compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/TypeReference.kt

```diff
@@ -50,7 +50,7 @@ import kotlin.LazyThreadSafetyMode.NONE
 @ExperimentalAnvilApi
 public sealed class TypeReference {
 
-  public abstract val declaringClass: ClassReference
+  public abstract val declaringClass: ClassReference?
```

Comment:
I had to change this for `TypeReference` to share the class between member references and top-level references.